catalog/plugin/catalog-plugin-metacardingest-network/src/main/java/org/codice/ddf/catalog/plugin/metacard/util/AttributeHelper.java

```diff
@@ -0,0 +1,204 @@
+     * whose value is represented by the given string {@param parsedValue}. Returns null if the string
+     * could not be parsed based on the descriptor's underlying {@link ddf.catalog.data.AttributeType.AttributeFormat}.
+     *
+     * @param attributeDescriptor The descriptor holding the format to use for parsing.
+     * @param parsedValue         The value to be parsed.
+     * @return A new {@link Attribute} if parsing succeeded, or null otherwise.
+     */
+    public Attribute createAttribute(AttributeDescriptor attributeDescriptor, String parsedValue) {
+        try {
+            notNull(attributeDescriptor);
+            notNull(parsedValue);
+
+            if (attributeDescriptor.isMultiValued()) {
+                LOGGER.trace("Found multi-valued attribute descriptor: {}",
+                        attributeDescriptor.getName());
+                return createMultiValuedAttribute(attributeDescriptor.getName(), parsedValue);
+            }
+
+            Serializable newValue;
+            switch (attributeDescriptor.getType()
+                    .getAttributeFormat()) {
+
+            case INTEGER:
+                newValue = Integer.parseInt(parsedValue);
+                break;
+
+            case FLOAT:
+                newValue = Float.parseFloat(parsedValue);
+                break;
+
+            case DOUBLE:
+                newValue = Double.parseDouble(parsedValue);
+                break;
+
+            case SHORT:
+                newValue = Short.parseShort(parsedValue);
+                break;
+            case LONG:
+                newValue = Long.parseLong(parsedValue);
+                break;
+
+            case DATE:
+                Calendar calendar = DatatypeConverter.parseDateTime(parsedValue);
+                newValue = calendar.getTime();
+                break;
+
+            case BOOLEAN:
+                newValue = Boolean.parseBoolean(parsedValue);
+                break;
+
+            case BINARY:
+                newValue = parsedValue.getBytes(Charset.forName("UTF-8"));
+                break;
+
+            case OBJECT:
+            case STRING:
+            case GEOMETRY:
+            case XML:
+                newValue = parsedValue;
+                break;
+
+            default:
+                return null;
+            }
+
+            return new AttributeImpl(attributeDescriptor.getName(), newValue);
+
+        } catch (IllegalArgumentException e) {
+            return null;
+        }
+    }
+
+    private void loadUniqueDescriptors(Set<AttributeDescriptor> descriptorsToLoad,
+            /*mutable*/ Map<String, AttributeDescriptor> destination) {
+        descriptorsToLoad.stream()
+                .filter(descriptor -> !destination.containsKey(descriptor.getName()))
+                .forEach(descriptor -> destination.put(descriptor.getName(), descriptor));
+    }
+
+    private Attribute createMultiValuedAttribute(String name, String commaSeparatedList) {
```

Comment:
Design after refactoring is a bit different. Made sense to combine the two and let the descriptor speak for how the input string gets processed. Much cleaner now. Will need a follow-up review for confirmation.